SMTP flaky test

What/why:
Flaky test in our pipelines for enterprise.

The test fails from time to time with:

```go
--- FAIL: TestSmtpSend (0.05s)
    --- FAIL: TestSmtpSend/multiple_recipients,_multiple_messages (0.03s)
        smtp_test.go:306:
            	Error Trace:	/drone/src/pkg/services/notifications/smtp_test.go:306
            	Error:      	Not equal:
            	            	expected: [][]string{[]string{“RCPT TO:<[rcpt1@example.com](mailto:rcpt1@example.com)>“, “250 Received”}}
            	            	actual  : [][]string{[]string{“RCPT TO:<[rcpt2@example.com](mailto:rcpt2@example.com)>“, “250 Received”}}
            	            	
            	            	Diff:
            	            	--- Expected
            	            	+++ Actual
            	            	@@ -2,3 +2,3 @@
            	            	  ([]string) (len=2) {
            	            	-  (string) (len=27) “RCPT TO:<[rcpt1@example.com](mailto:rcpt1@example.com)>“,
            	            	+  (string) (len=27) “RCPT TO:<[rcpt2@example.com](mailto:rcpt2@example.com)>“,
            	            	   (string) (len=12) “250 Received”
            	Test:       	TestSmtpSend/multiple_recipients,_multiple_messages
```

Drone builds:
https://drone.grafana.net/grafana/grafana-enterprise/75259/2/7

--- a/go.mod
+++ b/go.mod
@@ -126,7 +126,7 @@ require (
 	github.com/matttproud/golang_protobuf_extensions v1.0.4 // @grafana/alerting-backend
 	github.com/microsoft/go-mssqldb v1.7.0 // @grafana/grafana-bi-squad
 	github.com/mitchellh/mapstructure v1.5.1-0.20231216201459-8508981c8b6c //@grafana/identity-access-team
-	github.com/mocktools/go-smtp-mock/v2 v2.3.0 // @grafana/grafana-backend-group
+	github.com/mocktools/go-smtp-mock/v2 v2.3.1 // @grafana/grafana-backend-group
 	github.com/modern-go/reflect2 v1.0.2 // @grafana/alerting-backend
 	github.com/mohae/deepcopy v0.0.0-20170929034955-c48cc78d4826 // @grafana/alerting-backend
 	github.com/mwitkow/go-conntrack v0.0.0-20190716064945-2f068394615f // @grafana/grafana-operator-experience-squad

--- a/go.work.sum
+++ b/go.work.sum
@@ -885,6 +885,8 @@ github.com/mitchellh/iochan v1.0.0 h1:C+X3KsSTLFVBr/tK1eYN/vs4rJcvsiLU338UhYPJWe
 github.com/mitchellh/mapstructure v1.4.1/go.mod h1:bFUtVrKA4DC2yAKiSyO/QUcy7e+RRV2QTWOzhPopBRo=
 github.com/mithrandie/readline-csvq v1.3.0 h1:VTJEOGouJ8j27jJCD4kBBbNTxM0OdBvE1aY1tMhlqE8=
 github.com/mithrandie/readline-csvq v1.3.0/go.mod h1:FKyYqDgf/G4SNov7SMFXRWO6LQLXIOeTog/NB97FZl0=
+github.com/mocktools/go-smtp-mock/v2 v2.3.1 h1:wq75NDSsOy5oHo/gEQQT0fRRaYKRqr1IdkjhIPXxagM=
+github.com/mocktools/go-smtp-mock/v2 v2.3.1/go.mod h1:h9AOf/IXLSU2m/1u4zsjtOM/WddPwdOUBz56dV9f81M=
 github.com/modocache/gover v0.0.0-20171022184752-b58185e213c5 h1:8Q0qkMVC/MmWkpIdlvZgcv2o2jrlF6zqVOh7W5YHdMA=
 github.com/montanaflynn/stats v0.7.0 h1:r3y12KyNxj/Sb/iOE46ws+3mS1+MZca1wlHQFPsY/JU=
 github.com/mostynb/go-grpc-compression v1.2.2 h1:XaDbnRvt2+1vgr0b/l0qh4mJAfIxE0bKXtz2Znl3GGI=

--- a/pkg/services/notifications/smtp_test.go
+++ b/pkg/services/notifications/smtp_test.go
@@ -294,16 +294,16 @@ func TestSmtpSend(t *testing.T) {
 
 		count, err := client.Send(ctx, msgs...)
 		require.NoError(t, err)
-		require.Equal(t, 3, count)
+		assert.Equal(t, 3, count)
 
 		// workaround for https://github.com/mocktools/go-smtp-mock/issues/181
 		time.Sleep(1 * time.Millisecond)
 		messages := srv.MessagesAndPurge()
-		require.Len(t, messages, 3)
+		assert.Len(t, messages, 3)
 
 		for i, sentMsg := range messages {
 			rcpts := sentMsg.RcpttoRequestResponse()
-			require.EqualValues(t, [][]string{
+			assert.EqualValues(t, [][]string{
 				{fmt.Sprintf("RCPT TO:<rcpt%d@example.com>", i+1), "250 Received"},
 			}, rcpts)
 
@@ -316,7 +316,7 @@ func TestSmtpSend(t *testing.T) {
 			// make sure the trace is propagated
 			traceId := span.SpanContext().TraceID().String()
 			hasPrefix := strings.HasPrefix(hdr.Get("traceparent"), "00-"+traceId+"-")
-			require.True(t, hasPrefix)
+			assert.True(t, hasPrefix)
 
 			// one of the lines should be the body we expect!
 			found := false
@@ -334,7 +334,7 @@ func TestSmtpSend(t *testing.T) {
 				}
 			}
 
-			require.True(t, found)
+			assert.True(t, found)
 		}
 	})
 }

